examples/api/lib/widgets/routes/flexible_route_transitions.0.dart

```diff
@@ -0,0 +1,226 @@
+
+  @override
+  DelegatedTransitionBuilder? get delegatedTransition => VerticalPageTransition._delegatedTransition;
+
+  @override
+  Color? get barrierColor => const Color(0x00000000);
+
+  @override
+  bool get barrierDismissible => false;
+
+  @override
+  String? get barrierLabel => 'Should be no visible barrier...';
+
+  @override
+  bool get maintainState => true;
+
+  @override
+  bool get opaque => false;
+
+  // Begin PageRoute.
+
+  @override
+  Duration get transitionDuration => const Duration(milliseconds: 2000);
+
+  @override
+  Widget buildPage(BuildContext context, Animation<double> animation, Animation<double> secondaryAnimation) {
+    return builder(context);
+  }
+
+  static Widget buildPageTransitions<T>(
+    ModalRoute<T> route,
+    BuildContext context,
+    Animation<double> animation,
+    Animation<double> secondaryAnimation,
+    Widget child,
+  ) {
+    return VerticalPageTransition(
+      primaryRouteAnimation: animation,
+      secondaryRouteAnimation: secondaryAnimation,
+      child: child,
+    );
+  }
+
+  // TODO(justinmc): No canTransitionTo needed? I want to be able to have
+  // back-to-back VerticalPageTransitions.
+
+  @override
+  Widget buildTransitions(BuildContext context, Animation<double> animation, Animation<double> secondaryAnimation, Widget child) {
+    return buildPageTransitions<T>(this, context, animation, secondaryAnimation, child);
+  }
+
+  // End PageRoute.
+}
+
+class VerticalPageTransition extends StatelessWidget {
+  VerticalPageTransition({
+    super.key,
+    required Animation<double> primaryRouteAnimation,
+    required this.secondaryRouteAnimation,
+    required this.child,
+  }) : _primaryPositionAnimation =
+           CurvedAnimation(
+                 parent: primaryRouteAnimation,
+                 curve: curve,
+                 reverseCurve: curve,
+               ).drive(_kBottomUpTween),
+      _secondaryPositionAnimation =
+           CurvedAnimation(
+                 parent: secondaryRouteAnimation,
+                 curve: curve,
+                 reverseCurve: curve,
+               )
+           .drive(_kTopDownTween);
+
+  // When this page is coming in to cover another page.
+  final Animation<Offset> _primaryPositionAnimation;
+
+  // When this page is being coverd by another page.
+  final Animation<Offset> _secondaryPositionAnimation;
+
+  /// Animation
+  final Animation<double> secondaryRouteAnimation;
+
+  /// The widget below this widget in the tree.
+  final Widget child;
+
+  static const Curve curve = Curves.decelerate;
+
+  static final Animatable<Offset> _kBottomUpTween = Tween<Offset>(
+    begin: const Offset(0.0, 1.0),
+    end: Offset.zero,
+  );
+
+  static final Animatable<Offset> _kTopDownTween = Tween<Offset>(
+    begin: Offset.zero,
+    end: const Offset(0.0, -1.0),
+  );
+
+  static Widget _delegatedTransition(BuildContext context, Animation<double> animation, Animation<double> secondaryAnimation, Widget? child) {
```

Comment:
Probably worth a comment here?

```
// When the VerticalTransitionPageRoute animates onto or off of the navigation
// stack, this transition is given to the route below it so that they animate in
// sync.
```